### PR TITLE
194 display models metadata

### PR DESCRIPTION
**Pull request by FadyCoding**
## Describe your changes
Added a documentation button that displays on hover the metadata of models.
## Screenshots (if appropriate):
image

## Issue ticket number and link
https://github.com/debiai/DebiAI/issues/194
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have increased the version number in the `frontend/package.json` and `backend/swagger.yaml` files
- [x] I have checked that Black, Flake8, Prettier and Cspell are passing

### CODE REVIEW

File: backend/swagger.yaml [hunk 1]
```diff
@@ -483,8 +483,8 @@ paths:
         404:
           description: model or project doesn't exist
 
-  /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/resultsDict:
-    post:
+  ? /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/resultsDict
+  : post:
       summary: Add a results to a model
```
Comment (Tomansion) [inline backend/swagger.yaml:488]: To remove

File: backend/swagger.yaml [hunk 2]
```diff
@@ -524,8 +524,8 @@ paths:
         404:
           description: model or project doesn't exist
 
-  /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/getModelResults:
-    post:
+  ? /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/getModelResults
+  : post:
```
Comment (Tomansion) [inline backend/swagger.yaml:528]: To remove

File: backend/swagger.yaml [hunk 3]
```diff
@@ -706,8 +706,8 @@ paths:
           schema:
             type: object
 
-  /data-providers/{dataProviderId}/projects/{projectId}/selections/{selectionId}:
-    get:
+  ? /data-providers/{dataProviderId}/projects/{projectId}/selections/{selectionId}
+  : get:
```
Comment (Tomansion) [inline backend/swagger.yaml:710]: To remove

File: frontend/src/components/debiai/project/Models.vue [hunk 4]
```diff
@@ -152,6 +156,7 @@ export default {
       searchBar: "",
       selectedModelIds: [],
       commonModelResults: true,
+      metadata: null,
```
Comment (Tomansion) [inline frontend/src/components/debiai/project/Models.vue:159]: Is this used by something ?

File: frontend/src/services/services.js [hunk 5]
```diff
@@ -84,6 +84,20 @@ export default {
...
+      if (jsonObj.hasOwnProperty(key)) {
+        prettifiedString += `${key} : ${jsonObj[key]}\n`;
+      }
+    }
+    // Remove trailing newline character if needed
+    if (prettifiedString.endsWith("\n")) {
+      prettifiedString = prettifiedString.slice(0, -1);
+    }
+    return prettifiedString;
+  },
```
Comment (Tomansion) [inline frontend/src/services/services.js:99]: Does this function support nested json ?
The issue ticket dict : 
image

File: frontend/src/components/debiai/project/Models.vue [hunk 6]
```diff
@@ -194,6 +199,13 @@ export default {
         });
       event.stopPropagation();
     },
+    getStyles() {
+      return {
+        position: "relative",
+        left: "150px",
+        bottom: "-20px",
+      };
+    },
```
Comment (Tomansion) [inline frontend/src/components/debiai/project/Models.vue:208]: No need to put in a function if no complex logic.

**Review by Tomansion — CHANGES_REQUESTED** [review]
Some changes required :+1:

File: frontend/src/services/services.js (hunk 5 above)
Comment (FadyCoding) [reply frontend/src/services/services.js]: Does debiAI create_model function support nested json ?

File: frontend/src/components/debiai/project/Models.vue (hunk 4 above)
Comment (FadyCoding) [reply frontend/src/components/debiai/project/Models.vue]: Thank you for your comment, it has been removed.

File: frontend/src/services/services.js (hunk 5 above)
Comment (FadyCoding) [reply frontend/src/services/services.js]: It does not. I will implement a recursive version of the function to do so.

Comment (Tomansion) [reply frontend/src/services/services.js]: The metadata parameter of the create_model() function should, yes

File: backend/modules/dataProviders/webDataProvider/useCases/models.py [hunk 7]
```diff
@@ -8,19 +8,26 @@ def get_models_info(url, project_id):
         models = api.get_models(url, project_id)
         debiai_models = []
         for model_in in models:
-            # TODO: Deal with error if no model or no id
+            if "id" not in model_in:
+                continue
             model = {
                 "id": model_in["id"],
+                "metadata": {},
```
Comment (Tomansion) [inline backend/modules/dataProviders/webDataProvider/useCases/models.py:15]: Not necessary anymore

File: backend/swagger.yaml [hunk 8]
```diff
@@ -524,7 +524,7 @@ paths:
         404:
           description: model or project doesn't exist
 
-  /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/getModelResults:
+  /data-providers/{dataProviderId}/projects/{projectId}/models/{modelId}/getResults:
```
Comment (Tomansion) [inline backend/swagger.yaml:527]: Aren't you breaking something ?
Changing an API route url is a bold move

**Review by Tomansion — CHANGES_REQUESTED** [review]
Some few details

**Comment by Tomansion** [comment]
Nice work so far but I detected some issues with some specific metadata values: 
```json
{
      searchBar: "",
      selectedModelIds: [],
      commonModelResults: true,
      testJsonMetadata: {
        test: "test",
        array: [1, 2, 3],
        array2: [
          {
            test: "test",
            array: [1, 2, 3],
          },
          "test",
          1,
          29279827,
        ],
      },
```
Gives: 
image

This will need fixing

**Review by Tomansion — APPROVED** [review]
Well done ( ͡ᵔ ͜ʖ ͡ᵔ) :+1:

**Pull request merged**